packages/vsce/src/SshConfigUtils.ts

```diff
@@ -58,7 +145,234 @@ export class SshConfigUtils {
+            return undefined;
+        }
+        const sshRegex = /^ssh\s+([a-zA-Z0-9_-]+)@([a-zA-Z0-9.-]+)/;
+        const flagRegex = /--(\w+)(?:\s+("[^"]+"|'[^']+'|\S+))?/g;
+
+        const sshMatch = sshResponse.match(sshRegex);
+        if (!sshMatch) {
+            vscode.window.showErrorMessage("Invalid SSH command format. Ensure it matches the expected pattern.");
+            return undefined;
+        }
+        SshProfile.user = sshMatch[1];
+        SshProfile.hostname = sshMatch[2];
+
+        let flagMatch;
+        while ((flagMatch = flagRegex.exec(sshResponse)) !== null) {
+            const [, flag, value] = flagMatch;
+            const normalizedFlag = flag.toLowerCase();
+
+            // Check for missing value
+            if (!value) {
+                vscode.window.showErrorMessage(`Missing value for flag --${flag}.`);
+                return undefined;
+            }
+
+            const unquotedValue = value.replace(/^["']|["']$/g, ""); // Remove surrounding quotes
+
+            // Map aliases to consistent keys
+            if (normalizedFlag === "p" || normalizedFlag === "port") {
+                const portNumber = Number.parseInt(unquotedValue, 10);
+                if (Number.isNaN(portNumber)) {
+                    vscode.window.showErrorMessage(`Invalid value for flag --${flag}. Port must be a valid number.`);
+                    return undefined;
+                }
+                SshProfile.port = portNumber;
+            } else if (normalizedFlag === "pk" || normalizedFlag === "privateKey") {
+                SshProfile.privateKey = unquotedValue;
+            }
+
+            // Validate if quotes are required
+            if (/\s/.test(unquotedValue) && !/^["'].*["']$/.test(value)) {
+                vscode.window.showErrorMessage(`Invalid value for flag --${flag}. Values with spaces must be quoted.`);
+                return undefined;
+            }
+        }
+
+        if (!SshProfile.privateKey) {
+            if (privateKeyPath == null) {
+                privateKeyPath = true;
+            }
+        } else {
+            privateKeyPath = undefined;
+        }
+        try {
+            if (typeof privateKeyPath === "string") {
+                let match;
+                if ((match = /~(\/.*)/.exec(privateKeyPath))) {
+                    privateKeyPath = path.join(homedir(), match[1]);
+                }
+            } else if (privateKeyPath === true) {
+                for (const algo of ["id_ed25519", "id_rsa"]) {
+                    const tempPath = path.resolve(homedir(), ".ssh", algo);
+                    if (fs.existsSync(tempPath)) {
+                        privateKeyPath = path.resolve(homedir(), ".ssh", algo);
+                        break;
+                    }
+                }
+                if (SshProfile.privateKey == null) {
+                    throw Error("Failed to discover an ssh private key inside `~/.ssh`.");
+                }
```

Comment:
Should we move this to a public method like `findPrivateKey` in the SDK package?